src/main/scala/firrtl/transforms/DeadCodeElimination.scala

```diff
@@ -316,17 +316,17 @@ class DeadCodeElimination extends Transform {
     // Preserve original module order
     val newCircuit = c.copy(modules = c.modules.flatMap(m => modulesxMap.get(m.name)))
 
-    state.copy(circuit = newCircuit, renames = Some(renames))
+    state.copy(circuit = newCircuit, metadata = state.metadata.copy(renames = Some(renames)))
```

Comment:
we could add `def annotations: Seq[Annotation] = metadata.annotations` to CircuitState, which would simplify this syntax as well.